src/Surfnet/ServiceProviderDashboard/Application/Service/ServiceConnectionService.php

```diff
@@ -177,4 +179,42 @@ private function assertValidMetadata(ManageEntity $entity): MetaData
         }
         return $metadata;
     }
+
+    /**
+     * @return array<string, IdentityProvider>
+     */
+    private function getRemainingIdpConnections(InstitutionId $institutionId)
+    {
+        $items = $this->idpService->findInstitutionIdps($institutionId);
+        $indexed = [];
+        foreach ($items->institutionEntities() as $idp) {
+            $indexed[$idp->getEntityId()] = $idp;
+        }
+        return $indexed;
+    }
+
+    /**
+     * @param array<string, IdentityProvider> $testIdpsIndexed
+     * @return array<string, IdentityProvider>
+     */
+    private function gatherConnectedIdps(ManageEntity $entity, array $testIdpsIndexed): array
+    {
+        $connectedIdps = [];
+        foreach ($entity->getAllowedIdentityProviders()->getAllowedIdentityProviders() as $identityProvider) {
+            if (array_key_exists($identityProvider, $testIdpsIndexed)) {
+                $connectedIdps[$identityProvider] = $testIdpsIndexed[$identityProvider];
+            }
+        }
+        return $connectedIdps;
+    }
+
+    private function getTeam(MetaData $metadata): string
+    {
+        $team = $metadata->getCoin()->getServiceTeamId();
```

Comment:
Not that it is a biggy, but the method you call returns an Id (I think) than it would be clearer to use $teamId and even an opportunity to use a VO here. No blocker